### DISCUSSION TITLE
fepx ctest failed

### DISCUSSION THREAD

**Discussion by hongzzhang** [neperfepx/FEPX#99] [Q&A]
After setup, running the ctest, the result is follows, 
93% tests passed, 2 tests failed out of 30

Total Test time (real) = 106.77 sec

The following tests FAILED:
          5 - 05_dual_bcc_fcc (Failed)
         21 - 21_triaxial_clr (Failed)
Errors while running CTest
Output from these tests are in: /home/zhang/Downloads/FEPX/src/build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
make: *** [Makefile:71: test] Error 8

how can I deal with these errors? Pleae give advices. Thanks.

**Comment by mkasemer** [comment]
@hongzzhang I need more information to diagnose. Are you running the same computational setup - OS, compiler, MPI - that is described in the manual? If not, we would not expect that the test would succeed. At a bare minimum, please provide the above versions.

**Reply by hongzzhang** [reply]
@mkasemer many thanks for your response. I have run according to the same computational setup. The detailed information is as follows,
Distributor ID: Ubuntu, Description:  Ubuntu 22.04.1 LTS, Release: 22.04, Codename:  jammy;  GNU Fortran (Ubuntu 11.2.0-19ubuntu1) 11.2.0; mpirun (Open MPI) 4.1.4.

**Reply by mkasemer** [reply]
@hongzzhang - this is not the same computational setup as described in the [documentation](https://fepx.info/doc/introduction.html#testing-fepx). As such, we don't necessarily expect that the results will be similar, and thus the test may fail.

The question is does the simulation fail, or is it simply minor differences in output? If the former, it is likely a compiler issue, and currently we only support up to the compiler described in the documentation.

**EDIT** As a note, if these simulations are truly failing, you can attempt to debug and patch yourself and submit the fix via a pull request. Otherwise, likely our next earliest upgrade will be to Ubuntu 24.04 on or after April 2024, at which time we will begin focusing on whatever compiler versions are available at that time.

**Reply by hongzzhang** [reply]
@mkasemer thanks again. I will run the same computational setup as described in the documentation. If any problems, I will let you know.

**Reply by rquey** [reply]
@hongzzhang With your current setup, you should still

#### Run the tests in [Minimal mode](https://fepx.info/doc/introduction.html#testing-fepx):
```
cmake -DBUILD_TESTING_MODE=Minimal ..
ctest
```
which will tell you whether FEPX fails altogether or if the output differs from the reference output.

If all tests pass, then 

#### Run the tests in [Normal mode](https://fepx.info/doc/introduction.html#testing-fepx):
```
cmake -DBUILD_TESTING_MODE=Normal ..
ctest
```
and, if some tests fail, look at how the output differs from the reference output.  When a test fails, a `FAILED` file is generated *in the failed test's directory* (if `05_dual_bcc_fcc` fails, then go to `tests/05_dual_bcc_fcc`) that tells you which files differ. Open and compare these files to see if there are major differences of if (more likely) only a few digits differ.

**Reply by hongzzhang** [reply]
Hi, everyone. I run the same computational setup as described in the documentation.  Fepx is all right. Thanks for your help.
